quantum/action_tapping.c

```diff
@@ -504,8 +483,28 @@ void waiting_buffer_scan_tap(void) {
     // invalid state: tapping_key released && tap.count == 0
     if (!tapping_key.event.pressed) return;
 
+#    if (defined(AUTO_SHIFT_ENABLE) && defined(RETRO_SHIFT))
+    uint16_t tapping_keycode = get_record_keycode(&tapping_key, false);
+#    endif
     for (uint8_t i = waiting_buffer_tail; i != waiting_buffer_head; i = (i + 1) % WAITING_BUFFER_SIZE) {
-        if (IS_TAPPING_KEY(waiting_buffer[i].event.key) && !waiting_buffer[i].event.pressed && WITHIN_TAPPING_TERM(waiting_buffer[i].event)) {
+        // clang-format off
+        if (IS_TAPPING_KEY(waiting_buffer[i].event.key) && !waiting_buffer[i].event.pressed && (
+            WITHIN_TAPPING_TERM(waiting_buffer[i].event)
+#    if defined(AUTO_SHIFT_ENABLE) && defined(RETRO_SHIFT)
+            || (
```

Comment:
This being missing made so much weird behavior and caused 2-3 months of on-and-off playing with one bug and then getting frustrated and quitting for a while.